src/sentry/snuba/metrics/query.py

```diff
@@ -286,6 +293,15 @@ def validate_end(self) -> None:
             raise InvalidParams("start must be before end")
 
     def validate_granularity(self) -> None:
+        # Logic specific to how we handle time series in discover
+        if self.use_case_key == UseCaseKey.PERFORMANCE and self.include_series:
+            if self.interval is not None:
+                if self.granularity.granularity > self.interval:
+                    for granularity in METRICS_LAYER_GRANULARITIES:
+                        if granularity < self.interval:
+                            object.__setattr__(self, "granularity", Granularity(granularity))
```

Comment:
am i miss something here? why don't you just use `setattr` [link](https://docs.python.org/3/library/functions.html#setattr)?
>setattr(object, name, value)
This is the counterpart of [getattr()](http://docs.python.org/library/functions.html#getattr). The arguments are an object, a string and an arbitrary value. The string may name an existing attribute or a new attribute. The function assigns the value to the attribute, provided the object allows it. For example, setattr(x, 'foobar', 123) is equivalent to x.foobar = 123.

Reply:
https://github.com/getsentry/sentry/pull/39919#discussion_r994226715